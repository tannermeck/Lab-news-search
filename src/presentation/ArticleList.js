import PropTypes from 'prop-types';
import React from 'react';
import { Article } from './Article';

export const ArticleList = ({ articles }) => {
    const articleList = articles.map(({ author, title, description, publishedAt }) => (
        <li key={`${author}-${title}-${publishedAt}`}>
            <Article author={author}
            title={title}
            description={description}
            />
        </li>
    ))
    return <ul>
        {articleList}
        </ul>
}
ArticleList.proptypes = {
    articles: PropTypes.arrayOf(
        PropTypes.shape({
            author: PropTypes.string,
            title: PropTypes.string.isRequired,
            description: PropTypes.string.isRequired
        })
    )
}