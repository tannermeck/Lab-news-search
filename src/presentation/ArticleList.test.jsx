import React from 'react';
import { render } from '@testing-library/react'
import { ArticleList } from './ArticleList';


describe('should render an article list', () => {
    it('it should render an array of authors, titles, and descriptions', () => {
        const articles = [ 
            {title:'bitcoin', author:'Elon Musk', description: 'Bitcoin soars'},
            {title:'health', author:'Dr. Phil', description:'stop eating red meats'}]
        const { asFragment } = render(
        <ArticleList  articles={articles} />);
        expect(asFragment()).toMatchSnapshot()
        
    })
})