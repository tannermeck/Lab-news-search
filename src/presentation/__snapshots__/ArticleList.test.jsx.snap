// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an article list it should render an array of authors, titles, and descriptions 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <p
        aria-label="author-tag"
      >
        Author: Elon Musk
      </p>
      <p>
        Title: bitcoin
      </p>
      <p>
        Description: Bitcoin soars
      </p>
    </li>
    <li>
      <p
        aria-label="author-tag"
      >
        Author: Dr. Phil
      </p>
      <p>
        Title: health
      </p>
      <p>
        Description: stop eating red meats
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
